src/Plugin/BotCommunicationHandler.php

```diff
@@ -62,154 +75,174 @@
         elseif($packet instanceof ChannelDeletePacket) $this->handleChannelDelete($packet);
         elseif($packet instanceof ChannelPinsUpdatePacket) $this->handleChannelPinsUpdate($packet);
+        elseif($packet instanceof ThreadCreatePacket) $this->handleThreadCreate($packet);
+        elseif($packet instanceof ThreadUpdatePacket) $this->handleThreadUpdate($packet);
+        elseif($packet instanceof ThreadDeletePacket) $this->handleThreadDelete($packet);
         elseif($packet instanceof RoleCreatePacket) $this->handleRoleCreate($packet);
         elseif($packet instanceof RoleUpdatePacket) $this->handleRoleUpdate($packet);
         elseif($packet instanceof RoleDeletePacket) $this->handleRoleDelete($packet);
         elseif($packet instanceof InviteCreatePacket) $this->handleInviteCreate($packet);
         elseif($packet instanceof InviteDeletePacket) $this->handleInviteDelete($packet);
         elseif($packet instanceof BanAddPacket) $this->handleBanAdd($packet);
         elseif($packet instanceof BanRemovePacket) $this->handleBanRemove($packet);
-        elseif($packet instanceof ServerJoinPacket) $this->handleServerJoin($packet);
-        elseif($packet instanceof ServerLeavePacket) $this->handleServerLeave($packet);
-        elseif($packet instanceof ServerUpdatePacket) $this->handleServerUpdate($packet);
-        elseif($packet instanceof DiscordDataDumpPacket) $this->handleDataDump($packet);
-        elseif($packet instanceof DiscordReadyPacket) $this->handleReady();
-    }
-
-    private function handleReady(): void{
-        //Default activity, Feel free to change activity after ReadyEvent.
-        $ac = new Activity(VersionInfo::NAME." v".VersionInfo::BASE_VERSION." | DiscordBot ".\JaxkDev\DiscordBot\VERSION, Activity::TYPE_PLAYING);
-        $this->plugin->getApi()->updateBotPresence($ac, Member::STATUS_IDLE)->otherwise(function(ApiRejection $a){
+        elseif($packet instanceof GuildJoinPacket) $this->handleGuildJoin($packet);
+        elseif($packet instanceof GuildLeavePacket) $this->handleGuildLeave($packet);
+        elseif($packet instanceof GuildUpdatePacket) $this->handleGuildUpdate($packet);
+        elseif($packet instanceof WebhooksUpdatePacket) $this->handleWebhooksUpdate($packet);
+        elseif($packet instanceof DiscordReadyPacket) $this->handleReady($packet);
+        elseif($packet instanceof BotUserUpdatePacket) $this->handleBotUserUpdate($packet);
+        else $this->plugin->getLogger()->warning("Unhandled packet: " . get_class($packet));
+    }
+
+    private function handleReady(DiscordReadyPacket $packet): void{
+        //Default activity, Feel free to change in event / later time
+        $ac = Activity::create(VersionInfo::NAME . " v" . VersionInfo::BASE_VERSION . " | DiscordBot " . \JaxkDev\DiscordBot\VERSION, ActivityType::GAME);
+
+        $event = new DiscordReadyEvent($this->plugin, $packet->getBotUser(), $ac, Status::IDLE);
+        $event->call();
+
+        //TODO, Commands here.
```

Comment:
Commands will require a lot of checking against existing commands since they persist over shutdown in discord, so to reduce API spam I think its best for plugin to gather all commands at start (with child plugins registering them onLoad() to API so API can instantly send the full list on bot start and the Bot host (internal&external) can them register / update / delete as appropriate based off that list and already existing commands.

Reply:
Target 3.1